### PR TITLE
Playing Cards Improvement

### PR DESCRIPTION
**Pull request by Anasari**
A small rework of the polaris card to make it better and easier to use. 

What this PR does:
**Add:** 
1. Deal multiple cards function, you can deal up to 10 cards (or how many cards there are in the deck)
2. Clicking on a deck with empty hand automatically draw cards. It will add it to your existing hand
3. Deck is now picked up by drag-click, code from paper-bin
4. Card shuffle cooldown (15 ticks)
5. Port in card shuffle sound from paradise.
6. Shuffle verb for the playing card
7. Ability to discard multiple card (one by one due to limitation of my coding skills)
8. Ability to right-click remove one card from a hand.
9. Cards from different deck cannot be mixed anymore. Exception for Spaceball and Cardemon

**Remove:**
1. Drag-click card dealing because it was tedious

**Changed:**
1. Hand of Card in hand that's concealed no longer visible if you examine it. No cheating / peeking. This is to allow for people to take / draw someone else's card and give it to them without knowing the contents.

I'm also thinking of adding in instructions to the description that you can pick up the card by dragging it over your sprite

Feedback highly welcome!

Screenshots:

### CODE REVIEW

**Comment by Anasari** [comment]
Added in multi-card discard and shuffle verb, and also added in an exception for pick-up so it work as intended.

Now the only thing I could think of is to add in a feature to remove cards individually from a hand without discarding them. Unfortunately my very limited coding skills mean I cannot do that at the moment.

**Comment by Anasari** [comment]
Should be feature-complete now.

**Comment by Anewbe** [comment]
I'll try to review this soon, sorry about the wait.

File: code/modules/games/cards.dm [hunk 1]
```diff
@@ -52,32 +53,47 @@
...
-		for(var/datum/playingcard/P in H.cards)
-			cards += P
-		qdel(O)
-		user << "You place your cards on the bottom of \the [src]."
-		return
+		if(H.parentdeck == src)
+			for(var/datum/playingcard/P in H.cards)
+				cards += P
+			qdel(H)
+			user << "You place your cards on the bottom of \the [src]."
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 2]
```diff
@@ -52,32 +53,47 @@
...
+			return
+		else
+			user << "You can't mix card(s) from other deck!"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.
Also, other **decks**, if you could.

File: code/modules/games/cards.dm [hunk 3]
```diff
@@ -52,32 +53,47 @@
...
 	set desc = "Draw a card from a deck."
 	set src in view(1)
 
+	var/mob/living/carbon/user = usr
+
 	if(usr.stat || !Adjacent(usr)) return
 
-	if(!istype(usr,/mob/living/carbon))
+	if(user.hands_are_full()) // Safety check lest the card disappear into oblivion
+		usr << "Your hands are full!"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 4]
```diff
@@ -52,32 +53,47 @@
...
+	if(!istype(usr,/mob/living/carbon))
+		return
 
 	if(!cards.len)
 		usr << "There are no cards in the deck."
 		return
 
 	var/obj/item/weapon/hand/H = user.get_type_in_hands(/obj/item/weapon/hand)
+	if(H && !(H.parentdeck == src))
+		user << "You can't mix card(s) from other deck!"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.
Also, other **decks**, if you could.

File: code/modules/games/cards.dm [hunk 5]
```diff
@@ -113,21 +130,51 @@
...
-	H.update_icon()
+	set category = "Object"
+	set name = "Deal Multiple Cards"
+	set desc = "Deal multiple cards from a deck."
+	set src in view(1)
+
+	if(usr.stat || !Adjacent(usr)) return
+
+	if(!cards.len)
+		usr << "There are no cards in the deck."
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 6]
```diff
@@ -143,36 +190,81 @@
...
+		if(H.parentdeck == src.parentdeck) // Prevent cardmixing
+			for(var/datum/playingcard/P in cards)
+				H.cards += P
+			H.concealed = src.concealed
+			user.drop_from_inventory(src)
+			qdel(src)
+			H.update_icon()
+			return
+		else
+			user << "You cannot mix card(s) from other deck!"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.
Also, other **decks**, if you could.

File: code/modules/games/cards.dm [hunk 7]
```diff
@@ -143,36 +190,81 @@
...
+/obj/item/weapon/deck/verb/verb_shuffle()
+	set category = "Object"
+	set name = "Shuffle"
+	set desc = "Shuffle the cards in the deck."
+	set src in view(1)
+	shuffle()
+
+/obj/item/weapon/deck/proc/shuffle()
+	var/mob/living/user = usr
+	if (cooldown < world.time - 15) // 15 ticks cooldown
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: Maybe drop this to 10?

File: code/modules/games/cards.dm [hunk 8]
```diff
@@ -143,36 +190,81 @@
...
+	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
+		if(!istype(usr, /mob/living/simple_animal))
+			if( !usr.get_active_hand() )		//if active hand is empty
+				var/mob/living/carbon/human/H = user
+				var/obj/item/organ/external/temp = H.organs_by_name["r_hand"]
+
+				if (H.hand)
+					temp = H.organs_by_name["l_hand"]
+				if(temp && !temp.is_usable())
+					user << "<span class='notice'>You try to move your [temp.name], but cannot!</span>"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 9]
```diff
@@ -143,36 +190,81 @@
...
+					return
+
+				user << "<span class='notice'>You pick up [src].</span>"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 10]
```diff
@@ -143,36 +190,81 @@
...
+	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
+		if(!istype(usr, /mob/living/simple_animal))
+			if( !usr.get_active_hand() )		//if active hand is empty
+				var/mob/living/carbon/human/H = user
+				var/obj/item/organ/external/temp = H.organs_by_name["r_hand"]
+
+				if (H.hand)
+					temp = H.organs_by_name["l_hand"]
+				if(temp && !temp.is_usable())
+					user << "<span class='notice'>You try to move your [temp.name], but cannot!</span>"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 11]
```diff
@@ -143,36 +190,81 @@
...
+					return
+
+				user << "<span class='notice'>You pick up [src].</span>"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/cards.dm [hunk 12]
```diff
@@ -241,11 +343,48 @@
...
+	set name = "Remove card"
+	set desc = "Remove a card from the hand."
+	set src in view(1)
+
+	var/mob/living/carbon/user = usr
+
+	if(user.stat || !Adjacent(user)) return
+
+	if(user.hands_are_full()) // Safety check lest the card disappear into oblivion
+		usr << "Your hands are full!"
```
Comment (Anewbe) [inline code/modules/games/cards.dm]: to_chat(user, "Message"), please.

File: code/modules/games/tarot.dm [hunk 13]
```diff
@@ -24,14 +24,20 @@
...
-		if(prob(50))
-			P.name += " reversed"
-		newcards += P
-		cards -= P
-	cards = newcards
-	user.visible_message("\The [user] shuffles [src].")
\ No newline at end of file
+/obj/item/weapon/deck/tarot/shuffle()
+	var/mob/living/user = usr
+	if (cooldown < world.time - 15)
```
Comment (Anewbe) [inline code/modules/games/tarot.dm]: Maybe drop this to 10, too?

**Comment by Anasari** [comment]
It seems I dun goofed heavily. I'll have to make a new PR. Sorry about this!

**Pull request closed without merging**